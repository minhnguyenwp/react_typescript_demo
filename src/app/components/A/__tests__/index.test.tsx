import * as React from 'react';
import { render } from '@testing-library/react';

import { A } from '../index';
import { themes } from 'styles/theme/themes';
import { DefaultTheme } from 'styled-components';

const renderWithTheme = (theme?: DefaultTheme) =>
  render(<A theme={theme || themes.light} />);

describe('<A />', () => {
  it('shoudl match snapshot', () => {
    expect(renderWithTheme).toMatchSnapshot();
  });

  it('should render an <a> tag', () => {
    const a = renderWithTheme();
    expect(a.container.querySelector('a')).toBeInTheDocument();
  });

  // you can only test the styles dedined inside styled components, not outside as props
  // check here: https://github.com/testing-library/jest-dom/issues/295
  // and here: https://github.com/testing-library/jest-dom/issues/113#issuecomment-496971128
  it('should have theme', () => {
    const a = renderWithTheme();
    expect(a.container.firstChild).toHaveStyle(`text-decoration: none;`);
  });
});
