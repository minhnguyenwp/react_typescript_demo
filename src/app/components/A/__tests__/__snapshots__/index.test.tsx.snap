// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<A /> shoudl match snapshot 1`] = `[Function]`;
